db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/FromBaseTable.java

```diff
@@ -1813,13 +1896,32 @@ else if(isOneRowResultSet()){
          * Splice changed the below loop to iterate over storeRestrictionList instead of restrictionList */
         for(int i=0;i<storeRestrictionList.size();i++){
             Predicate pred=storeRestrictionList.elementAt(i);
-            if(pred.isInListProbePredicate() && pred.isStartKey()){
-                disableBulkFetch();
-                multiProbing=true;
-                break;
+            if(pred.isStartKey()){
+                if (pred.isInListProbePredicate()) {
+                    disableBulkFetch();
+                    multiProbing = true;
+                }
+            }
+            if (pred.isStartKey() || pred.isStopKey()){
+                if (numUnusedLeadingIndexFields == 0) {
+                    numUnusedLeadingIndexFields = ap.getNumUnusedLeadingIndexFields();
+                    if (numUnusedLeadingIndexFields >= 1) {
+                        currentIndexFirstColumnStats.setFrom(trulyTheBestConglomerateDescriptor.getFirstColumnStats());
+                        if (numUnusedLeadingIndexFields > 1)
+                            throw StandardException.newException(LANG_INTERNAL_ERROR,
+                               "IndexPrefixIteratorMode currently allows at most one leading index column to be unspecified in predicates.");
```

Comment:
that's also what I thought when I saw some assumptions above in the code about considering only the first column, it is a good start to consider only the first column, nevertheless we should probably consider extending it to cover the first N-no-predicate-index-columns to unlock its full potential.

Reply:
Right, though the benefit will decrease the more index prefix values we have, and might at some crossover point be more expensive than doing a full table scan.  So, it is safer to start with 1 column, and then when we have more time to validate the costing and benefit of the multicolumn case, add it in later.